stable/minio/templates/statefulset.yaml

```diff
@@ -55,6 +55,34 @@ spec:
         - name: {{ .Chart.Name }}
           image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
           imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.gateway.enabled  }}
+          {{- if eq .Values.gateway.type "s3" }}
+          command: [ "/bin/sh",
+          "-ce",
+          "/usr/bin/docker-entrypoint.sh minio -C {{ .Values.configPath }} gateway s3 {{ .Values.gateway.s3.serviceEndpoint }}" ]
+          {{- end }}
+          {{- if eq .Values.gateway.type "azure" }}
+          command: [ "/bin/sh",
+          "-ce",
+          "/usr/bin/docker-entrypoint.sh minio -C {{ .Values.configPath }} gateway azure" ]
+          {{- end }}
+          {{- if eq .Values.gateway.type "gcs" }}
+          command: [ "/bin/sh",
+          "-ce",
+          "/usr/bin/docker-entrypoint.sh minio -C {{ .Values.configPath }} gateway gcs {{ .Values.gateway.gcs.projectId }}" ]
+          {{- end }}
+          {{- if eq .Values.gateway.type "oss" }}
+          command: [ "/bin/sh",
+          "-ce",
+          "cp /tmp/config.json {{ .Values.configPath }} &&
+          /usr/bin/docker-entrypoint.sh minio -C {{ .Values.configPath }} gateway oss {{ .Values.gateway.oss.endpointURL }}" ]
+          {{- end }}
+          {{- if eq .Values.gateway.type "nas" }}
+          command: [ "/bin/sh",
+          "-ce",
+          "/usr/bin/docker-entrypoint.sh minio -C {{ .Values.configPath }} gateway nas {{ .Values.mountPath }}" ]
+          {{- end }}
+          {{- else }}
```

Comment:
We can probably use StatefulSets always for gateways, but  PVC need to be mounted on value specified in `MINIO_CACHE_DRIVES`  for this to work properly

Reply:
Thank for your comment, I added this env for gateway with persistence=enabled

Reply:
Thanks @galiev will test in a bit